packages/lodestar/src/chain/chain.ts

```diff
@@ -169,11 +169,31 @@ export class BeaconChain implements IBeaconChain {
     const currentEpochStartSlot = computeStartSlotAtEpoch(this.clock.currentEpoch);
     const head = this.forkChoice.getHead();
     const bestSlot = currentEpochStartSlot > head.slot ? currentEpochStartSlot : head.slot;
-    return await this.regen.getBlockSlotState(head.blockRoot, bestSlot);
+    const mChildLabels = {entrypointFn: "getBlockSlotState", callingModule: "getProposerDuties"};
+    return await this.regen.getBlockSlotState(
+      head.blockRoot,
+      bestSlot,
+      this.metrics
+        ? {
+            stateLookupsTotal: this.metrics.regenStateCacheLookupTotal.child(mChildLabels),
+            stateLookupHits: this.metrics.regenStateCacheLoopkupHits.child(mChildLabels),
+          }
+        : undefined
+    );
   }
 
   async getHeadStateAtCurrentSlot(): Promise<CachedBeaconState<allForks.BeaconState>> {
-    return await this.regen.getBlockSlotState(this.forkChoice.getHeadRoot(), this.clock.currentSlot);
+    const mChildLabels = {entrypointFn: "getBlockSlotState", callingModule: "getProposerDuties"};
+    return await this.regen.getBlockSlotState(
```

Comment:
Here you are telling the "getBlockSlotState" function that it's own name is "getBlockSlotState". That information should already be known, no need to pass it as an argument.

Reply:
The function only needs to know who is it calling it. There's not need to pass an entire metrics instance, you can just pass the name of the callingModule as an enum
```ts
function getBlockSlotState (blockRoot: Root, slot: Slot, caller?: GetBlockSlotStateCaller) {}

enum GetBlockSlotStateCaller {
  getProposerDuties = "getProposerDuties"
}
```